File: assets/api/add_method.py
import urllib2
import json
import random

# this is a sample python client for the JSON API for adding a method
# author: Jiri Kadlec

username = 'admin'
password = 'password'

methID = int(random.random() * 1000)
methName = "Python Method %s" % methID

#STEP 1: prepare the uploaded data in JSON format
data = {
    "user": username,
    "password": password,
    "MethodDescription": methName,
    "MethodLink": "http://example.com",
    "VariableID": 0
}
postdata = json.dumps(data)


#STEP 2: post the data values to HydroServer and check response status
uploadURL = 'http://worldwater.byu.edu/app/index.php/default/services/api/methods'
req = urllib2.Request(uploadURL)
req.add_header('Content-Type', 'application/json')

try:
    response = urllib2.urlopen(req, postdata)
    print response.read()

except urllib2.HTTPError, e:
    print e.code
    print e.msg
    print e.headers
    print e.fp.read()

File: assets/api/add_method.R
#you must install the packages httr and RJSONIO.
library(httr)
library(RJSONIO)

random_id = round(runif(1, 0, 1000))
random_name = paste("R method", random_id)
random_code = paste("R", random_id, sep="-")

x <- list(
  user = "admin",
  password = "password",
  MethodDescription = random_name,
  MethodLink = "http://example.com",
  VariableID = 0
)

response <- POST("http://worldwater.byu.edu/app/index.php/default/services/api/methods",
                 body = RJSONIO::toJSON(x),
                 add_headers("Content-Type" = "application/json"),
                 verbose()
)

response_status = content(response, type="application/json")
print(response_status)
